### ISSUE TITLE
Components and needed  permission

### ISSUE THREAD

**Issue by pMertDogan** [mit-cml/appinventor-sources#1879]
**Describe the desired feature**
We cant see permission thats needed for components.

Some components needs special permissions.Like phone component.But we cant see this permission on development.We can see this permision request when we install app to device or when we open screen that screen uses this component.

Another reason we cant idenfity which component need this special permissions.ıf our project old and big.Its hard to find it.

**Give an example of how this feature would be used**
Maybe there is a permission tab and we can see components and requered permissionso we can try alternative components or we dont use it.

Students understand permissions on Android ecosystem.Rules/Policy etc.

**Why doesn't the current App Inventor system address this use case?**
We cant see needed permissions for each component

No confilics
**Why is this feature beneficial to App Inventor's educational mission?**

Its expand  students view.They can see how permissions work on Android

**Comment by ewpatton** [comment]
I’m weary of this because the whole point of MIT App Inventor is to lower the barrier to entry of making app development; having to ask app inventors to reason about the permission model is probably asking too much for most of them. Also, the permission model on Android and iOS are slightly different and we don’t necessarily want to make platform-specific allowances in the web UI as we move forward with our cross platform support.

**Comment by ewpatton** [comment]
This is also complicated by the fact that some permissions are now enabled on a per-property/per-block basis. It wouldn’t just be on the component level put on each block potentially.

**Comment by ewpatton** [comment]
@SusanRatiLane I'd like to think about how we can turn this into a GSOC project to build a UI that explains what's going on and possibly allows people to retract permissions, with appropriate warnings.

**Comment by basvi-chunara** [comment]
Hello @ewpatton @SusanRatiLane,
I was wondering if this issue is available for GSOC 2026?

**Comment by SusanRatiLane** [comment]
> Hello [@ewpatton](https://github.com/ewpatton) [@SusanRatiLane](https://github.com/SusanRatiLane), I was wondering if this issue is available for GSOC 2026?

Yes, it is. You're welcome to submit a proposal. Be sure to review our guidance for this year: https://community.appinventor.mit.edu/t/google-summer-of-code-gsoc-2026/

**Comment by basvi-chunara** [comment]
Thanks for the confirmation!

**Comment by Pranav140** [comment]
> > Hello [@ewpatton](https://github.com/ewpatton) [@SusanRatiLane](https://github.com/SusanRatiLane), I was wondering if this issue is available for GSOC 2026?
> 
> Yes, it is. You're welcome to submit a proposal. Be sure to review our guidance for this year: https://community.appinventor.mit.edu/t/google-summer-of-code-gsoc-2026/

I'd like to contribute to this issue. After analyzing the codebase, I see the infrastructure 
is already in place to build a practical solution:

**Current State**:
- `ComponentInfo.getPermissionsNeeded()` maps components → permissions
- `LoadComponentInfo.generatePermissions()` already aggregates component metadata
- Components declare permissions via `@UsesPermissions` annotations (e.g., `BluetoothClient`)
- `PermissionRegistry.java` tracks SDK version constraints

**Proposed Solution**:
1. **Backend**: Extend `ProjectService.java` RPC to expose `getProjectPermissionMetadata()` 
   - Reuse existing `LoadComponentInfo` permission aggregation
   - Build DTO: `PermissionManifest` (Component → Permission → SDK constraints)

2. **Frontend**: Add PermissionsPanel tab in `YaProjectEditor.java`
   - Hierarchical tree: Component → Required Permissions → SDK versions
   - Shows which components use which permissions
   - Displays min SDK requirements

3. **Advanced**: Handle per-block permissions (e.g., `Web.PostFile` only needs `WRITE_EXTERNAL_STORAGE`)
   - Extend permission parsing to check method-level `@UsesPermissions`

**Implementation Scope**: ~200 LOC backend + ~400 LOC frontend (backend: RPC + data aggregation, 
frontend: TreeWidget + constraint visualization)

**Benefits**: Students understand Android permissions upfront, can choose alternatives, and learn 
the permission model directly supporting the educational mission.

This aligns with the GSOC 2026 project scope. I'm ready to open a design document and discuss 
any concerns about cross-platform support or architecture decisions.

Interested in working on this. Feedback welcome!

**Comment by SusanRatiLane** [comment]
> I'd like to contribute to this issue.

If you would like feedback on a proposal, please post it to our community forum https://community.appinventor.mit.edu/c/open-source-development/gsoc/26

**Comment by Pranav140** [comment]
> > I'd like to contribute to this issue.
> 
> If you would like feedback on a proposal, please post it to our community forum https://community.appinventor.mit.edu/c/open-source-development/gsoc/26

Hello , this is the PoC for the issue . I will submit be proposal ASAP.

**Comment by SyedAhad01** [comment]
Hi, I would like to work on this feature.

I plan to start by extracting permissions from components using the @UsesPermissions annotation and verifying them in the backend.

Then I will explore how to display this information in the UI, possibly starting with a simple output and later improving it into a dedicated panel.

Please let me know if this approach sounds good.

**Comment by Pranav140** [comment]
> > I'd like to contribute to this issue.
> 
> If you would like feedback on a proposal, please post it to our community forum https://community.appinventor.mit.edu/c/open-source-development/gsoc/26

I have posted the proposal in the Proposal reviews topic in the community . 

Looking forward for your guidance regarding it.
